### ISSUE TITLE
[Bug] Failed to write A response payload

### ISSUE THREAD

**Issue by KipRire** [MetaCubeX/mihomo#806]
### Verify steps

- [X]  确保你使用的是**本仓库**最新的的 clash 或 clash Alpha 版本 Ensure you are using the latest version of Clash or Clash Premium from **this repository**. 
- [ ]  如果你可以自己 debug 并解决的话，提交 PR 吧 Is this something you can **debug and fix**? Send a pull request! Bug fixes and documentation fixes are welcome. 
- [X]  我已经在 [Issue Tracker](……/) 中找过我要提出的问题 I have searched on the [issue tracker](……/) for a related issue. 
- [X]  我已经使用 Alpha 分支版本测试过，问题依旧存在 I have tested using the dev branch, and the issue still exists. 
- [X]  我已经仔细看过 [Documentation](https://wiki.metacubex.one/) 并无法自行解决问题 I have read the [documentation](https://wiki.metacubex.one/) and was unable to solve the issue. 
- [X]  这是 Clash 核心的问题，并非我所使用的 Clash 衍生版本（如 OpenClash、KoolClash 等）的特定问题 This is an issue of the Clash core *per se*, not to the derivatives of Clash, like OpenClash or KoolClash. 

### Clash version

v1.16.0, 2.8.0Meta

### What OS are you seeing the problem on?

Windows, Linux

### Clash config

```yaml
Here's my proxy config:

- {name: Reality, server: web.of.reality.com, port: 443, type: vless, uuid: 0000-0000-0000-0000-0000, network: tcp, tls: true, udp: true, flow: xtls-rprx-vision, servername: web.of.reality.com, reality-opts: {public-key: my_public_key, short-id: abc}, client-fingerprint: chrome}
```

### Description

Failed to write A response payload
---
I Got this **`[warning] app/proxyman/inbound: connection ends > proxy/vless/inbound: connection ends > proxy/vless/inbound: failed to write A response payload > write unix /dev/shm/xray.sock->@: write: broken pipe`** in xray `error.log` while using clash.meta's [vless-reality-vision](https://wiki.metacubex.one/config/proxies/vless/#vless-reality-vision). 
1. Every time this "warning" pump up the internet interruption occured, or *vice versa*.  Cannot visit websites smoothly (In my case: google search, youtube, cloudflare homepage, facebook).
2. I've tested that this situation can be found both on clash.meta-core **v1.16.0**, clash.meta for **android** and clash.meta-core **Alpha**(Oct 17 12:52:30 CST 2023). And not found in **v2rayN(v6.2.7 with xray-core1.8.4)**.
    - [windows] clash.meta v1.16.0:                                      Got warning
    - [windows] clash.meta prereliease-alpha OCT-17:        Got warning
    - [Android] clash.meta for android 2.8.9Meta:               Got warning
    - [windows] v2rayN (xray-core 1.8.4)                              No warning
4. Since it's just a "warning", so I do not turn on the "show" button of xray-reality-server-end for further analysis.
5. My reality config refers to [lxhao61 integrated-examples](https://github.com/lxhao61/integrated-examples/tree/main/Xray(M%2BF%2BB%2BG%2BA)%2BNginx) which use Nginx's SNI proxying to Xray, i.e. "steal_yourself"
    - web.of.reality.com --> nginx-stream --> xray.sock --> Xray-server --> nginx.sock --> my_website

Besides, I got this answer from Claude2:
```
So in summary, a VLESS client made a request to Xray, which was processed successfully, but when trying to send the response back, the client connection was broken/closed so Xray couldn't send the data.
This is usually caused by the client closing the connection for some reason before reading the response. It could be due to a timeout, client crash, or some other issue.
```
